cpp/arcticdb/pipeline/write_frame.cpp

```diff
@@ -325,23 +329,23 @@ static RowRange partial_rewrite_row_range(
     }
 }
 
-folly::Future<std::optional<SliceAndKey>> async_rewrite_partial_segment(
+folly::Future<SliceAndKey> async_rewrite_partial_segment(
         const SliceAndKey& existing, const IndexRange& index_range, VersionId version_id,
         AffectedSegmentPart affected_part, const std::shared_ptr<Store>& store
 ) {
     return store->read(existing.key())
             .thenValueInline(
                     [existing, index_range, version_id, affected_part, store](
                             std::pair<VariantKey, SegmentInMemory>&& key_segment
-                    ) -> folly::Future<std::optional<SliceAndKey>> {
+                    ) -> folly::Future<SliceAndKey> {
                         const auto& key = existing.key();
                         const SegmentInMemory& segment = key_segment.second;
                         const RowRange affected_row_range =
                                 partial_rewrite_row_range(segment, index_range, affected_part);
                         const auto num_rows = int64_t(affected_row_range.end() - affected_row_range.start());
-                        if (num_rows <= 0)
-                            return std::nullopt;
-
+                        if (num_rows <= 0) {
+                            return folly::Try<SliceAndKey>{}; // Empty future
```

Comment:
How does this affect the callers of `async_rewrite_partial_segment`? It seems like the output type is changed but I don't see any changes to the code using `async_rewrite_partial_segment`.